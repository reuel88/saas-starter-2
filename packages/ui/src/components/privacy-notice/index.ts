export { PrivacyNotice } from "./privacy-notice";
