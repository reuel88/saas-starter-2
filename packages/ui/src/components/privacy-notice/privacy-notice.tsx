"use client";

import { FC, useEffect, useRef, useState } from "react";
import { useCookie } from "../../hooks";
import { Button } from "../button";
import { Container } from "../container";
import { Wrapper } from "../wrapper";

export type PrivacyNoticeProps = {};

const COOKIE_NAME = "privacy-notice";

export const PrivacyNotice: FC<PrivacyNoticeProps> = () => {
  const ref = useRef<HTMLDivElement>(null);
  const [height, setHeight] = useState(0);
  const [cookieValue, setCookie] = useCookie(COOKIE_NAME, false);

  useEffect(() => {
    if (!ref) return;

    setHeight(ref.current?.offsetHeight || 0);
  });

  const handleAcceptAll = () => {
    setCookie(true);
  };

  const handleRejectAll = () => {
    setCookie(true, { expires: 1 });
  };

  return cookieValue ? null : (
    <>
      <Wrapper
        ref={ref}
        className="fixed inset-x-0 bottom-0 border-t border-border bg-background py-6"
      >
        <Container className="flex flex-col justify-between gap-x-8 gap-y-4 md:flex-row md:items-center">
          <p className="text-background-foreground max-w-4xl text-sm leading-6">
            This website uses cookies to supplement a balanced diet and provide
            a much deserved reward to the senses after consuming bland but
            nutritious meals. Accepting our cookies is optional but recommended,
            as they are delicious. See our{" "}
            <a
              href="#"
              className="font-semibold text-hue-700 dark:text-hue-400"
            >
              cookie policy
            </a>
            .
          </p>
          <div className="flex flex-none items-center gap-x-5">
            <Button type="button" onClick={handleAcceptAll}>
              Accept all
            </Button>
            <Button type="button" variant="outline" onClick={handleRejectAll}>
              Reject all
            </Button>
          </div>
        </Container>
      </Wrapper>
      <div style={{ height }} />
    </>
  );
};
