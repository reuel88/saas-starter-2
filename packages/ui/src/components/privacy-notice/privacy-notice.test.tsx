import { render } from "@testing-library/react";
import { PrivacyNotice } from "./privacy-notice";

describe("PrivacyNotice", () => {
  it("should render a div container with children when no 'as' prop is passed", () => {
    // Arrange

    // Act
    const { baseElement } = render(<PrivacyNotice />);

    // Assert
    expect(baseElement).toMatchSnapshot();
  });
});
