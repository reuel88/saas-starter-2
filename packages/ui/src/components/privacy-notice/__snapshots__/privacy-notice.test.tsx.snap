// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PrivacyNotice > should render a div container with children when no 'as' prop is passed 1`] = `
<body>
  <div>
    <div
      class="text-foreground fixed inset-x-0 bottom-0 border-t border-border bg-background py-6"
    >
      <div
        class="mx-auto max-w-7xl px-6 lg:px-8 flex flex-col justify-between gap-x-8 gap-y-4 md:flex-row md:items-center"
      >
        <p
          class="text-background-foreground max-w-4xl text-sm leading-6"
        >
          This website uses cookies to supplement a balanced diet and provide a much deserved reward to the senses after consuming bland but nutritious meals. Accepting our cookies is optional but recommended, as they are delicious. See our
           
          <a
            class="font-semibold text-hue-700 dark:text-hue-400"
            href="#"
          >
            cookie policy
          </a>
          .
        </p>
        <div
          class="flex flex-none items-center gap-x-5"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
            type="button"
          >
            Accept all
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
            type="button"
          >
            Reject all
          </button>
        </div>
      </div>
    </div>
    <div
      style="height: 0px;"
    />
  </div>
</body>
`;
